engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/JobPreparationActor.scala

```diff
@@ -20,76 +22,83 @@ case class JobPreparationActor(executionData: WorkflowExecutionActorData,
       attributesWithDefault = curriedAddDefaultsToAttributes(evaluatedRuntimeAttributes)
       jobDescriptor = BackendJobDescriptor(workflowDescriptor.backendDescriptor, jobKey, attributesWithDefault, inputs)
-    } yield BackendJobPreparationSucceeded(jobDescriptor, factory.jobExecutionActorProps(jobDescriptor, initializationData, serviceRegistryActor))
+    } yield BackendJobPreparationSucceeded(jobDescriptor, factory.jobExecutionActorProps(jobDescriptor, initializationData, serviceRegistryActor))) match {
+      case Success(s) => s
+      case Failure(f) => BackendJobPreparationFailed(jobKey, f)
+    }
+
   }
 
   // Split inputs map (= evaluated workflow declarations + coerced json inputs) into [init\.*].last
   private lazy val splitInputs = workflowDescriptor.backendDescriptor.inputs map { case (fqn, v) => splitFqn(fqn) -> v }
 
   def resolveAndEvaluateInputs(jobKey: BackendJobDescriptorKey,
-                               wdlFunctions: WdlStandardLibraryFunctions): Try[Map[LocallyQualifiedName, WdlValue]] = {
+                               wdlFunctions: WdlStandardLibraryFunctions): Future[Try[Map[LocallyQualifiedName, WdlValue]]] = {
     import RuntimeAttributeDefinition.buildMapBasedLookup
-    val call = jobKey.call
-    lazy val callInputsFromFile = unqualifiedInputsFromInputFile(call)
-    lazy val workflowScopedLookup = hierarchicalLookup(jobKey.call, jobKey.index) _
-
-    // Try to resolve, evaluate and coerce declarations in order
-    val inputEvaluationAttempt = call.task.declarations.foldLeft(Map.empty[LocallyQualifiedName, Try[WdlValue]])((inputs, declaration) => {
-      val name = declaration.name
-
-      // Try to resolve the declaration, and upon success evaluate the expression
-      // If the declaration is resolved but can't be evaluated this will throw an evaluation exception
-      // If it can't be resolved it's ignored and won't appear in the final input map
-      val evaluated: Option[Try[WdlValue]] = declaration.expression match {
-        // Static expression in the declaration
-        case Some(expr) => Option(expr.evaluate(buildMapBasedLookup(inputs), wdlFunctions))
-        // Expression found in the input mappings
-        case None if call.inputMappings.contains(name) => Option(call.inputMappings(name).evaluate(workflowScopedLookup, wdlFunctions))
-        // Expression found in the input file
-        case None if callInputsFromFile.contains(name) => Option(Success(callInputsFromFile(name)))
-        // Expression can't be found
-        case _ => None
-      }
-
-      // Leave out unresolved declarations
-      evaluated match {
-        case Some(value) =>
-          val coercedValue = value flatMap declaration.wdlType.coerceRawValue
-          inputs + ((name, coercedValue))
-        case None => inputs
-      }
-    })
-
-    TryUtil.sequenceMap(inputEvaluationAttempt, s"Input evaluation for Call ${call.fullyQualifiedName} failed")
+    Future {
```

Comment:
Why is this actor now running yet another `Future` inside the akka `receive` (that is already inside a `Future` itself)? If conceptually this method should create a second `Future`, then why return a `Future[Try[A]]` instead of just a flattened `Future[A]`.